### PR TITLE
Update metadata-extensions

### PR DESCRIPTION
**Pull request by tvdijen**
Part of the v5 release of saml2 is that some of the new work from master has been backported.
This PR fixes the instantiation and parsing of the md-extensions.

### CODE REVIEW

**Comment by tvdijen** [comment]
Still gotta do md:ContactPerson and md:KeyDescriptor.. Was a little bit premature to open this PR

File: src/SimpleSAML/Metadata/SAMLBuilder.php [hunk 1]
```diff
@@ -173,157 +173,66 @@ public function addSecurityTokenServiceType(array $metadata): void
...
-                $s = new Scope();
-                $s->setScope($scopetext);
-                // Check whether $ ^ ( ) * | \ are in a scope -> assume regex.
-                if (1 === preg_match('/[\$\^\)\(\*\|\\\\]/', $scopetext)) {
-                    $s->setIsRegexpScope(true);
-                } else {
-                    $s->setIsRegexpScope(false);
-                }
-                $e->setExtensions(array_merge($e->getExtensions(), [$s]));
+                $isScoped = (1 === preg_match('/[\$\^\)\(\*\|\\\\]/', $scopetext));
```
Comment (thijskh) [inline src/SimpleSAML/Metadata/SAMLBuilder.php:189]: `s/isScoped/isRegexpScope/`

File: src/SimpleSAML/Metadata/SAMLBuilder.php [hunk 2]
```diff
@@ -173,157 +173,66 @@ public function addSecurityTokenServiceType(array $metadata): void
      */
     private function addExtensions(Configuration $metadata, RoleDescriptor $e): void
     {
+        $extensions = [];
+
         if ($metadata->hasValue('hint.cidr')) {
```
Comment (thijskh) [inline src/SimpleSAML/Metadata/SAMLBuilder.php:178]: Maybe we should just drop the _undocumented_ `hint.cidr` option and instead only keep the DiscoHints IPHint option which does the same thing?

**Review by thijskh — APPROVED** [review]

**Comment by tvdijen** [comment]
@thijskh Would you mind going over this again? It's really done this time ;)

**Pull request merged**